koku/koku/pg_partition.py

```diff
@@ -1668,3 +1679,193 @@ def repartition_default_data(schema_name=None, partitioned_table_name=None):
+    default_params = {k: v for k, v in part_rec.items() if k in default_keys}
+    default_params["table_name"] = f"{default_params['partition_of_table_name']}_default"
+    default_params["partition_parameters"] = {"default": True}
+    default_part, created = PartitionedTable.objects.get_or_create(
+        schema_name=default_params["schema_name"],
+        table_name=default_params["table_name"],
+        partition_of_table_name=default_params["partition_of_table_name"],
+        partition_parameters__default=True,
+        defaults=default_params,
+    )
+
+    LOG.info(f"{'Created' if created else 'Retrieved'} default partition {default_part.table_name}")
+
+    return default_part, created
+
+
+def _check_default_partition_data(default_partition, part_rec):
+    """
+    Use the specified record to check if the default partition contains data
+    for the parameters defined in the partition record. If it detects data, the partition parameters
+    will be altered to be some time in the future for a range or a ramdom array of letters for a list.
+    The original parameters will be returned if this is the case. Also returned will be the
+    SQL where clause needed to check the conditions. This will be re-used when moving data.
+
+    Params:
+        default_partition (PartitionedTable) : The tracking record for the default partition
+        part_rec (dict) : A dict describing a partitioned_tables record.
+
+    Returns: (tuple)
+        (dict, str):
+            dict : The original partition parameters or empty dict
+            str : The SQL where clause (minus "where") for the needed conditions when moving data
+    """
+    params = None
+    if default_partition.partition_type == default_partition.RANGE:
+        chk_where = f'"{default_partition.partition_col}" >= %s AND "{default_partition.partition_col}" < %s'
+        params = [part_rec["partition_parameters"]["from"], part_rec["partition_parameters"]["to"]]
+    else:
+        chk_where = f'"{default_partition.partition_col}" IN ( %s )'
+        params = [part_rec["partition_parameters"]["in"]]
+
+    with transaction.get_connection().cursor() as cur:
+        chk_where = cur.mogrify(chk_where, params).decode("utf-8")
+
+    chk_sql = f"""
+SELECT EXISTS (
+    SELECT 1
+      FROM "{default_partition.schema_name}"."{default_partition.table_name}"
+     WHERE {chk_where}
+);
+"""
+    with transaction.get_connection().cursor() as cur:
+        cur.execute(chk_sql)
+        res = cur.fetchone()
+
+    if res[0]:
+        LOG.info(f"Overlapping data found in the default partition for {chk_where}")
+        restore = part_rec["partition_parameters"].copy()
+        if default_partition.partition_type == default_partition.RANGE:
+            pp_from = ciso8601.parse_datetime(part_rec["partition_parameters"]["from"]).date()
+            pp_to = ciso8601.parse_datetime(part_rec["partition_parameters"]["to"]).date()
+            delta = relativedelta(years=random.randrange(100, 7000))
```

Comment:
what is this for?

Reply:
@maskarb 
I was concerned that, if I used a constant, there might be a collision from concurrent processes.
I confess I might have been overthinking this, though.